### PR TITLE
#24134: Default logic app versions to ~4

### LINKED ISSUE #24134: azurerm_logic_app_standard default version should be ~4
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

Latest

### AzureRM Provider Version

3.83.0

### Affected Resource(s)/Data Source(s)

azurerm_logic_app_standard 

### Terraform Configuration Files

```hcl
n/a
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

azurerm_logic_app_standard default version should be ~4 instead of ~3. Extension version 4 went GA last October. 3 is deprecated. https://azure.microsoft.com/en-us/updates/logic-apps-standard-support-for-functions-runtime-v4/

### Actual Behaviour

azurerm_logic_app_standard default version is currently ~3

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by harshavmb**
Default logic app versions to ~4

Fixes #24134

### CODE REVIEW

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @harshavmb thanks for submitting this. This looks good, but we should deprecate this with a 4.0 feature flag so this won't cause issues for users currently relying on the default value of `~3`. This is an example of a Default value deprecation in another resource [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/ab4f143d37136a8b0b6a492de0e51178cb26e279/internal/services/iothub/iothub_resource.go#L222).

**Comment by harshavmb** [comment]
Hi @catriona-m ,

Thanks for the review & I amended changes as per the review. I do see all PRs failing `
Pull Request Triage / triage (pull_request_target)` check. I don't think it's related to my change.

File: website/docs/r/logic_app_standard.html.markdown [hunk 1]
```diff
@@ -151,7 +151,9 @@ The following arguments are supported:
 
 ~> **Note:**  When using an App Service Plan in the `Free` or `Shared` Tiers `use_32_bit_worker_process` must be set to `true`.
 
-* `version` - (Optional) The runtime version associated with the Logic App. Defaults to `~3`.
+* `version` - (Optional) The runtime version associated with the Logic App. Defaults to `~4`.
```
Comment (catriona-m) [inline website/docs/r/logic_app_standard.html.markdown:154]: we can leave this as `Defaults to `~3` for now until we officially move to 4.0.

**Review by catriona-m — APPROVED** [review]
Thanks @harshavmb LGTM!

**Pull request merged**